internal_testsuites/golang/testsuite/check_that_apic_network_is_private/check_that_apic_network_is_private_test.go

```diff
@@ -0,0 +1,82 @@
+package check_that_apic_network_is_private
+
+import (
+	"context"
+	"github.com/kurtosis-tech/kurtosis-cli/golang_internal_testsuite/test_helpers"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+const (
+	testName              = "apic-is-private"
+	isPartitioningEnabled = false
+	defaultDryRun         = false
+	defaultParallelism    = 4
+	emptyArgs             = "{}"
+
+	// This Script fails if the APIC has a non private IP address range
+	starlarkScript = `
```

Comment:
Having this as an E2E test seems like an overhead. I would rather have us unit test this or, even better, having `panic` + comment on the original function to make sure this precondition doesn't change without something bad happening